sentence_transformers/util.py

```diff
@@ -609,3 +917,68 @@ def get_device_name() -> Literal["mps", "cuda", "npu", "cpu"]:
         return "npu"
     else:
         return "cpu"
+
+
+def quantize_embeddings(
+    embeddings: Union[Tensor, np.ndarray],
+    precision: Literal["float32", "int8", "uint8", "binary", "ubinary"],
+    ranges: Optional[np.ndarray] = None,
+    calibration_embeddings: Optional[np.ndarray] = None,
+) -> np.ndarray:
+    """
+    Quantizes embeddings to a lower precision. This can be used to reduce the memory footprint and increase the
+    speed of similarity search. The supported precisions are "float32", "int8", "uint8", "binary", and "ubinary".
+
+    :param embeddings: Embeddings with dtype float32 to convert
+    :param precision: The precision to convert to. Options are "float32", "int8", "uint8", "binary", "ubinary".
+    :param ranges: Ranges for quantization of embeddings. This is only used for int8 quantization, where the ranges
+        refers to the minimum and maximum values for each dimension. So, it's a 2D array with shape (2, embedding_dim).
+        Default is None, which means that the ranges will be calculated from the calibration embeddings.
+    :type ranges: Optional[np.ndarray]
+    :param calibration_embeddings: Embeddings used for calibration during quantization. This is only used for int8
+        quantization, where the calibration embeddings can be used to compute ranges, i.e. the minimum and maximum
+        values for each dimension. Default is None, which means that the ranges will be calculated from the query
+        embeddings. This is not recommended.
+    :type calibration_embeddings: Optional[np.ndarray]
+    :return: Quantized embeddings with the specified precision
+    """
+    if isinstance(embeddings, Tensor):
+        embeddings = embeddings.cpu().numpy()
+    elif isinstance(embeddings, list):
+        if isinstance(embeddings[0], Tensor):
+            embeddings = [embedding.cpu().numpy() for embedding in embeddings]
+        embeddings = np.array(embeddings)
+    if embeddings.dtype != np.float32:
```

Comment:
This should work for float16 (and 64) as well no?

Reply:
Fixed in 54478bb563a8ac311d8d8f8b8dce9139b742956c